packages/hosted-credit-card-integration/src/components/HostedCreditCardComponent.tsx

```diff
@@ -0,0 +1,278 @@
+import { CardInstrument, LegacyHostedFormOptions } from '@bigcommerce/checkout-sdk';
+import { compact, forIn } from 'lodash';
+import React, { FunctionComponent, ReactNode, useCallback, useState } from 'react';
+
+import {
+    CreditCardPaymentMethodComponent,
+    CreditCardPaymentMethodProps,
+} from '@bigcommerce/checkout/credit-card-integration';
+import {
+    CreditCardCustomerCodeField,
+    CreditCardInputStylesType,
+    getCreditCardInputStyles,
+    isInstrumentCardCodeRequiredSelector,
+    isInstrumentCardNumberRequiredSelector,
+} from '@bigcommerce/checkout/instrument-utils';
+import { PaymentMethodProps } from '@bigcommerce/checkout/payment-integration-api';
+
+import { getHostedCreditCardValidationSchema } from './getHostedCreditCardValidationSchema';
+import { getHostedInstrumentValidationSchema } from './getHostedInstrumentValidationSchema';
+import { HostedCreditCardFieldset } from './HostedCreditCardFieldset';
+import { HostedCreditCardValidation } from './HostedCreditCardValidation';
+
+const HostedCreditCardComponent: FunctionComponent<PaymentMethodProps> = ({
+    method,
+    checkoutService,
+    checkoutState,
+    paymentForm,
+    language,
+    onUnhandledError,
+}) => {
+    const [focusedFieldType, setFocusedFieldType] = useState<string>();
+
+    const { setFieldTouched, setFieldValue, setSubmitted, submitForm } = paymentForm;
+    const isInstrumentCardCodeRequiredProp = isInstrumentCardCodeRequiredSelector(checkoutState);
+    const isInstrumentCardNumberRequiredProp =
+        isInstrumentCardNumberRequiredSelector(checkoutState);
+    const {
+        config: { cardCode, showCardHolderName },
+    } = method;
+    const isCardCodeRequired = cardCode || cardCode === null;
+    const isCardHolderNameRequired = showCardHolderName ?? true;
+
+    const getHostedFieldId: (name: string) => string = useCallback(
```

Comment:
I see `useCallback` is being used a lot in this component.
Could you please explain are they really improving the performance here?

Reply:
I just moved code from parent component to a new one. It was written with useCallbacks before me

Reply:
Could you please do a quick to test on if removing some or all `useCallback` has any impact on re-render times of ` <CreditCardPaymentMethodComponent />`? 
As I am working on converting class components into function components, I noticed some `useCallback` usages do not make a difference.

Reply:
Peng asked me to measure the performance of the component and you can see that component with useCallback for now is better based on numbers. I think useCallback should be moved in React 19 migration project